### PR TITLE
Unify `*ApiImplWithAuthorization` and "base" `*ApiImpl` classes

### PR DESCRIPTION
**Pull request by snazy**
This change simplifies the code and makes authorization a "first class" citizen. Also prevents duplicate calls to `namedRefWithHashOrThrow` for `getContent`+`getContents`+`getDiff`. Introduces a `NoopAuthorizer` for the case that there's no authorization.

### CODE REVIEW

File: servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java [hunk 1]
```diff
@@ -228,6 +235,22 @@ public Reference getReferenceByName(String refName, FetchOption fetchOption)
...
+    BatchAccessChecker check =
+        startAccessCheck().canCreateReference(RefUtil.toNamedRef(type, refName));
+    try {
+      check.canViewReference(namedRefWithHashOrThrow(sourceRefName, targetHash).getValue());
+    } catch (NessieNotFoundException e) {
+      // If the default-branch does not exist and hashOnRef points to the "beginning of time",
+      // then do not throw a NessieNotFoundException, but re-create the default branch. In all
+      // cases, re-throw the exception.
+      if (!(Reference.ReferenceType.BRANCH.equals(type)
+          && refName.equals(getConfig().getDefaultBranch())
```
Comment (dimas-b) [inline servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java:247]: So we do not run the default branch re-creation requests through access checks? Is that intended?

Would it be preferable to call `namedRefWithHashOrThrow` (line 241) separately and always pass the branch name to `check.canViewReference()`?

**Review by dimas-b — APPROVED** [review]
LGTM. The comments apply to old code as well, so it's more of a general "should we fix this?" question.

**Comment by snazy** [comment]
Added notes for your comments to #5926

File: servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java (hunk 1 above)
Comment (snazy) [reply servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java:247]: The "hard check" is intentional.

**Pull request merged**